File: backend/src/main/java/com/altstay/api/amenity/Amenity.java
package com.altstay.api.amenity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "amenity")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Amenity {

    @Id
    @Column(name = "code", nullable = false)
    private String code;

    @Column(name = "label", nullable = false)
    private String label;

    @Column(name = "category", nullable = false)
    private String category;
}

File: frontend/src/lib/contracts/amenity.ts
import { z } from 'zod';

// Mirrors com.altstay.api.amenity.AmenityController.AmenityResponse. No nullable fields — every
// column backing this record is `not null`.
export const AmenityResponseSchema = z.object({
  code: z.string(),
  label: z.string(),
  category: z.string(),
});
export type AmenityResponse = z.infer<typeof AmenityResponseSchema>;
